index.js

```diff
@@ -1,51 +1,54 @@
 'use strict';
 
 var through = require('through2');
-var removeBom = require('remove-bom-buffer');
+var TextDecoder = require('util').TextDecoder;
 
-function removeBomStream() {
-  var state = 0; // 0:Not removed, -1:In removing, 1:Already removed
-  var buffer = Buffer.alloc(0);
+var BOM = Buffer.from([0xEF, 0xBB, 0xBF], 'utf-8');
 
-  return through(onChunk, onFlush);
+function removeBomStream(encoding) {
+  encoding = (encoding || '').toLowerCase();
+  var isUTF8 = (encoding === 'utf-8' || encoding === 'utf8');
 
-  function removeAndCleanup(data) {
-    state = 1; // Already removed
+  if (!isUTF8) {
+    return through();
+  }
 
-    buffer = null;
+  // Only used if encoding is UTF-8
+  var decoder = new TextDecoder('utf-8', { ignoreBOM: false });
 
-    return removeBom(data);
-  }
+  var state = 0; // 0:Not removed, -1:In removing, 1:Already removed
+
+  return through(onChunk);
 
-  function onChunk(data, enc, cb) {
+  function onChunk(data, _, cb) {
     if (state === 1) {
-      return cb(null, data);
+      cb(null, data);
+      return;
     }
 
-    if (state === 0 /* Not removed */ && data.length >= 7) {
-      return cb(null, removeAndCleanup(data));
-    }
+    try {
+      state = -1;
 
-    state = -1; // In removing
+      var chunk = decoder.decode(data, { stream: true });
 
-    var bufferLength = buffer.length;
-    var chunkLength = data.length;
-    var totalLength = bufferLength + chunkLength;
+      // The first time we have data after a decode, it should have already removed the BOM
+      if (chunk !== '') {
+        chunk += decoder.decode(); // end of stream mode and clear inner buffer.
 
-    buffer = Buffer.concat([buffer, data], totalLength);
+        var buffer = Buffer.from(chunk, 'utf-8');
 
-    if (totalLength >= 7) {
-      return cb(null, removeAndCleanup(buffer));
-    }
-    cb();
-  }
+        // Node<=v11, TextDecoder#decode returns a BOM if it receives a BOM separately.
```

Comment:
```suggestion
        // Node<=v11, TextDecoder#decode returns a BOM if it receives a BOM separately.
        // Ref https://github.com/nodejs/node/pull/30132
```

Reply:
By the way, this is still broken in nodejs, as they don't check the correct encodings. I might try to submit a patch, but we still need this implementation for now.